File: ios/Classes/Models/PeripheralState.swift
//
//  PeripheralStatus.swift
//  flutter_ble_peripheral
//
//  Created by Julian Steenbakker on 26/11/2021.
//

import Foundation

enum PeripheralState : Int{
//    case idle, unauthorized, unsupported, advertising, connected
    /// Status is not (yet) determined.
    case unknown

    /// BLE is not supported on this device.
    case unsupported

    /// BLE usage is not authorized for this app.
    case unauthorized

    /// BLE is turned off.
    case poweredOff

    // /// Android only: Location services are disabled.
    // locationServicesDisabled,

    /// BLE is fully operating for this app.
    case idle

    /// BLE is advertising data.
    case advertising

    /// BLE is connected to a device.
    case connected
    
//    var index: Int { PeripheralState..firstIndex(of: self) ?? 0 }
}

File: lib/src/core/enums/peripheral_state.dart
/*
 * Copyright (c) 2020. Julian Steenbakker.
 * All rights reserved. Use of this source code is governed by a
 * BSD-style license that can be found in the LICENSE file.
 */

/// The live advertising state, as published on the state change stream.
///
/// The index of each entry is the value the platform sends, so the order must
/// not change.
enum PeripheralState {
  /// Status is not (yet) determined.
  unknown,

  /// BLE is not supported on this device.
  unsupported,

  /// BLE usage is not authorized for this app.
  unauthorized,

  /// BLE is turned off.
  poweredOff,

  /// Android only: Location services are disabled.
  locationServicesDisabled,

  /// BLE is fully operating for this app.
  idle,

  /// BLE is advertising data.
  advertising,

  /// BLE is connected to a device.
  connected,

  /// Android only: the permission was denied once, so a rationale should be
  /// shown before asking again.
  shouldShowRequestPermissionRationale,
}
